oauth2client/contrib/gce.py

```diff
@@ -48,28 +45,67 @@
 """
 
 
-def _get_service_account_email(http_request=None):
-    """Get the GCE service account email from the current environment.
-
+def _get_metadata(http_request=None, path=None, recursive=True):
+    """Gets a JSON object from the specified path on the Metadata Server
     Args:
-        http_request: callable, (Optional) a callable that matches the method
-                      signature of httplib2.Http.request, used to make
-                      the request to the metadata service.
-
+        http_request: an httplib2.Http().request object or equivalent
+            with which to make the call to the metadata server
+        *path: a list of strings denoting the metadata server request
+            path.
     Returns:
-        tuple, A pair where the first entry is an optional response (from a
-        failed request) and the second is service account email found (as
-        a string).
+        A deserialized JSON object representing the data returned
+        from the metadata server
     """
-    if http_request is None:
+
+    if not path:
+        path = []
+
+    if not http_request:
         http_request = httplib2.Http().request
+
+    r_string = '/?recursive=true' if recursive else ''
+    full_path = _METADATA_ROOT + '/'.join(path) + r_string
     response, content = http_request(
-        _DEFAULT_EMAIL_METADATA, headers={'Metadata-Flavor': 'Google'})
+        full_path,
+        headers={'Metadata-Flavor': 'Google'}
+    )
     if response.status == http_client.OK:
-        content = _from_bytes(content)
-        return None, content
+        decoded = _from_bytes(content)
+        if response['content-type'] == 'application/json':
+            return json.loads(decoded)
+        else:
+            return decoded
     else:
-        return response, content
+        msg = (
+            'Failed to retrieve {path} from the Google Compute Engine'
+            'metadata service. Response:\n{error}'
+        ).format(path=full_path, error=response)
+        raise ValueError(msg)
+
+
+def _get_access_token(http_request, email):
+    """Get an access token for the specified email from the Metadata Server.
+    Args:
+        http_request: an httplib2.Http().request object or equivalent
+            with which to make the call to the metadata server
+        email: The service account email to request an access token with
+    Returns:
+        A tuple (accessToken, token expiry)
+    """
+    token_json = _get_metadata(
+        http_request=http_request,
+        path=[
+            'instance',
+            'service-accounts',
+            email,
+            'token'
+        ],
+        recursive=False
+    )
+    token_expiry = datetime.datetime.now() + datetime.timedelta(
+        seconds=token_json.get('expires_in')
+    )
+    return token_json.get('access_token'), token_expiry
```

Comment:
Ditto here on the `KeyError`. If `access_token` isn't there, then this method has failed big time?